lib/bambora/bank/batch_report_resource.rb

```diff
@@ -51,11 +53,20 @@ def initialize(client:, api_key:)
       #
       #  @params profile_data [Hash] with values as noted in the example.
       def show(report_data)
-        client.post(path: sub_path, body: batch_report_body(report_data))
+        add_messages_to_response(
+          client.post(path: sub_path, body: batch_report_body(report_data)),
+        )
       end
 
       private
 
+      def add_messages_to_response(response)
+        response.dig(:response, :record).map! do |record|
+          record.merge!(messages: record[:messageId].split(',').map { |id| MESSAGES[id] })
```

Comment:
Do you think this would be more intuitive if we name this `error_messages` instead of `messages`?

Reply:
The field from Bambora is `messageId` and not all of the messages are errors, so I think it's better to keep the field named `message`.

`BatchReportResource` is used in HiMama indirectly by calling `Bambora::Client.new(args).batch_reports(api_key)`